File: src/main/java/com/example/stopgap/exception/ExceptionEndpoint.java
package com.example.stopgap.exception;

import com.example.stopgap.Endpoint;
import com.example.stopgap.instanceregistry.InstanceRegistry;
import io.helidon.http.HttpException;
import io.helidon.http.Status;
import io.helidon.webserver.http.HttpRules;
import io.helidon.webserver.http.HttpService;

public final class ExceptionEndpoint implements Endpoint {

    @Override
    public HttpService routes(final InstanceRegistry registry) {
        return (final HttpRules rules) -> {
            rules
                .get("/bad-client-1", (req, res) -> {
                    throw new BadClientException();
                })
                .get("/bad-client-2", (req, res) -> {
                    throw new HttpException("Bad client request directly", Status.BAD_REQUEST_400);
                });
        };
    }

}

File: app/src/main/kotlin/com/example/stopgap/exception/ExceptionEndpoint.kt
package com.example.stopgap.exception

import dev.sku20.helidon.endpoint.Endpoint
import dev.sku20.helidon.endpoint.Get
import dev.sku20.helidon.endpoint.Post
import io.helidon.http.HttpException
import io.helidon.http.Status
import io.helidon.webserver.http.ServerRequest
import io.helidon.webserver.http.ServerResponse

@Endpoint("/exception")
class ExceptionEndpoint {

    @Get("/bad-client-1")
    fun badClientException(req: ServerRequest, res: ServerResponse) {
        throw BadClientException()
    }

    @Get("/bad-client-2")
    fun viaHttpException(req: ServerRequest, res: ServerResponse) {
        throw HttpException("Bad client request directly", Status.BAD_REQUEST_400)
    }

    @Post("/bad-post")
    fun post(req: ServerRequest, res: ServerResponse) {
        throw HttpException("Bad Request", Status.BAD_REQUEST_400)
    }

}
